corehq/apps/data_dictionary/views.py

```diff
@@ -72,19 +85,16 @@ def data_dictionary_json_case_types(request, domain):
         queryset = queryset.filter(is_deprecated=False)
 
     case_type_app_module_count = get_case_type_app_module_count(domain)
-    used_props_by_case_type = get_used_props_by_case_type(domain)
     geo_case_prop = get_geo_case_property(domain)
     case_types_data = []
     for case_type in queryset:
         module_count = case_type_app_module_count.get(case_type.name, 0)
-        used_props = used_props_by_case_type.get(case_type.name, [])
         case_types_data.append({
             "name": case_type.name,
             "fhir_resource_type": fhir_resource_type_name_by_case_type.get(case_type),
             "is_deprecated": case_type.is_deprecated,
             "module_count": module_count,
             "properties_count": case_type.properties_count,
-            "is_safe_to_delete": len(used_props) == 0,
```

Comment:
I'm a little confused about the changes to this function and the one below, but not to `data_dictionary_json`, which also has a `is_safe_to_delete` flag. It looks like the `is_safe_to_delete` is still present in the data dictionary js. I've only looked briefly but haven't figured out where `data_dictionary_json` is used.

Reply:
That wasn't a question. Is it intentional that you're not changing `data_dictionary_json`?

Reply:
Does https://github.com/dimagi/commcare-hq/pull/36295 answer your question? This PR is based on that branch and should swich to master once that is approved/merged.

Reply:
oh, yep, hadn't read that one. That clarifies most of my question, but I still don't understand why you didn't need to update [this js](https://github.com/dimagi/commcare-hq/blob/bc8a307c86fe596f0a8a99b457179ff256e674ea/corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js#L184) - isn't that property always going to be false now that `is_safe_to_delete` isn't being passed?

Reply:
Ah I see. Yeah I saw there was some logic based on a geo case type for showing/hiding the delete button in addition to used properties, so I left that property in. See [this change](https://github.com/dimagi/commcare-hq/pull/36297/commits/37c8a847bb96624fe5b6eb4cf8ea4e9ea421d2b5#diff-a9f010cc42e7d1be32f617e83127bb635d0fca956a6f846eb1dc2eb6f6b25f3eL163-R160). I don't love the subtle difference in behavior there, but also felt like changing that behavior was out of scope of this PR. I could be convinced that it is not though.